src/controllers/authController.js

```diff
@@ -158,9 +158,47 @@ const forgotPassword = async (req, res) => {
   res.status(StatusCodes.OK).json({ message: 'Password reset email sent' });
 };
 
+//reset Password endpoint
+const reset_password = async (req, res) => {
+  const { passwordToken, newPassword } = req.body;
+  //AKOS: How would we receive this passwordToken from the frontend?
+  // Find the user with the provided password token
+  const user = await User.findOne({
+    passwordResetToken: passwordToken,
+    passwordResetExpiresOn: { $gt: new Date() }, // Check if reset token is still valid
+  });
+
+  if (!user) {
+    throw new CustomError.NotFoundError('User not found');
+  }
+
+  // Check if password reset token is still valid
+  const now = new Date();
+  if (user.passwordResetExpiresOn < now) {
```

Comment:
This was already checked on line 168. The user would not have been returned from our mongo query if the `passwordResetExpiresOn` date was in the past. 

If you prefer to check them separately to give more granular error messages; then you can remove the `passwordResetExpiresOn` filter from the earlier query.

Reply:
@akosasante sure! wanted to do that but forgot to!

Reply:
I was following tickets's instructions and write explicit error messages- but I will leave two options in the code(1st one is commented out)